### ISSUE TITLE
How could I add a link for the "Edit" icon to private Bitbucket repo?

### ISSUE THREAD

**Issue by corinabioinformatic** [rstudio/bookdown#1435]
<img width="222" alt="topButtonEdit_bookdown" src="https://github.com/rstudio/bookdown/assets/14955321/646a1e99-64b9-46a7-8be3-14d673f2d657">

Hello, related to Bookdown I am editing the "index.Rmd" and "_output.yaml" file in order to make functional the "Edit" button. 
I followed the instructions detailed in this page: [29.4 Edits and source code](https://rstudio4edu.github.io/rstudio4edu-book/book-dress.html#edits-and-source-code)

But when render my book and I click on that "Edit icon" I get again the error page in Bitbucket. 
<img width="511" alt="error404_bitbucket" src="https://github.com/rstudio/bookdown/assets/14955321/22c85dd4-a6ad-4283-9478-69a8c52a3bdc">

I have a private bitbucket and I added the following lines of code in the :
"_output.yml"
```
bookdown::gitbook:
  css: style.css
  config:
    toc:
      collapse: section
      before: |
        <li><a href="./">Demo Book</a></li>
      after: |
        <li><a href="https://bitbucket.whatever.com/projects/my_repo_name/my_bookdown" target="blank">Published with bookdown</a></li>
    edit:
      link: bitbucket.whatever.com/my_username/my_repo_name/edit/master/%s
      text: "Suggest an edit"

    download: ["pdf", "epub"]
```

"index.Rmd"
```
title: "bookdown"
author:"authors"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography:
- book.bib
- packages.bib
description: 
  Thats it
link-citations: yes
always_allow_html: yes
github-repo: my_bitb_username/my_repo_name
```

Does anybody know if Bookdown can have a Edit button to edit in a private Bitbucket repo? (or suggest edits?)
Many thanks!

**Comment by cderv** [comment]
> But when render my book and I click on that "Edit icon" I get again the error page in Bitbucket.

What is the link supposed to be for bitbucket source file edition ? 

What would be the correct link to generate ?

**Comment by cderv** [comment]
What you described above does not match what you put in your configuration 
```
edit:
      link: bitbucket.whatever.com/my_username/my_repo_name/edit/master/%s
```

I believe Bitbucket is adding some parameters to URL  so you could try putting the same url you have in your browser and adds `?mode=edit&at=main` at the end. 

The URL I got on bitbucket.org is not exactly the same as yours though 
`````
https://bitbucket.org/dervieuxchr/bookdown-demo/src/main/README.md
`````
when I clicked it adds `mode=edit&at=main`

So you need to adapt and put the right url in `edit` config, leaving `%s` where the filename should go.

**Comment by ElenaMetori** [comment]
Hello, 

Same problem here but with the links to a devops repo

- the url for the project must be something like :
https://dev.azure.com/company_name/project_name/_git/repo_name 
- for a file (index.Rmd for example) : 
https://dev.azure.com/company_name/project_name/_git/repo_name?path=/index.Rmd 
(the url doesn't change on edit mode)

I've modified the links in the "_output.yml" : 
```
bookdown::bs4_book:
  css: css/style.css
  theme:
    primary: "#096B72"
  repo: https://dev.azure.com/company_name/project_name/_git/repo_name
  edit: https://dev.azure.com/company_name/project_name/_git/repo_name&path=/%s
```

But the links on the built book are modified as: 
- https://dev.azure.com/company_name/project_name/_git/repo_name/blob/master/index.Rmd
- https://dev.azure.com/company_name/project_name/_git/repo_name/edit/master/index.Rmd

**Comment by ElenaMetori** [comment]
I think the problem comes from the function `tweak_navbar()` in bs4_book.R : 

```
repo_edit <- paste0(repo$base, "/edit/", repo$branch, "/", repo$subdir, rmd_index[[active]])
repo_view <- paste0(repo$base, "/blob/", repo$branch, "/", repo$subdir, rmd_index[[active]])
```

**Comment by cderv** [comment]
yes bs4_book() does not support the url you want to put. If should be extended in configuration to allow that. 

This is probably part of 

* https://github.com/rstudio/bookdown/issues/1011

**Comment by ElenaMetori** [comment]
Any way to workaround other than make a local modified version of bookdown ?

**Comment by cderv** [comment]
Unfortunately know, you found the place where this is built
https://github.com/rstudio/bookdown/blob/cedaac98f2d4b54a97865e5b050a38fb8938cc7f/R/bs4_book.R#L416-L427

Currently it will add the file in the path, and not as a query params  like your dev azure expect. 

A PR if welcome to improve support.

**Comment by ElenaMetori** [comment]
#1447

**Comment by ElenaMetori** [comment]
I also modified the `template_link()` function in order to open the link into a new tab, but I'm not sure that you want a new PR with that too ?

**Comment by cderv** [comment]
> I also modified the `template_link()` function in order to open the link into a new tab, but I'm not sure that you want a new PR with that too ?

Yes you can make a PR for this too.
